### ISSUE TITLE
Support enum validation parameter

### ISSUE THREAD

**Issue by asieira** [nknapp/bootprint-swagger#18]
The `enum` keyword allows the list of valid values for a schema entry to be defined. It would be great if that was reflected on the documentation accordingly, by listing the different options automatically in the generated HTML.

**Comment by nknapp** [comment]
Can you be more specific about which enum objects you would expect to be displayed? 
I've just verified with the latest version, that enums are displayed in parameter  [see the operation /pet/findByStatus](http://www.knappi.org/bootprint-petstore/#path--pet-findByStatus) and schema-sections (see the "Order" object) on the same page.
Enums are displayed like `string ∈ { "placed" , "approved" , "delivered" }`

**Comment by asieira** [comment]
Ah, it seems that the problem is that I have an array with strings defined as enums.

In my example, I have a response or definition that is an object with a field with the following structure:
```
                                    "report_tags": {
                                        "type": "array",
                                        "items": {
                                            "type": "string",
                                            "enum": [
                                                "IPv4.P2P.generic",
                                                "IPv4.P2P.BitTorrent",
                                                "IPv4.P2P.Skype",
                                                "IPv4.TorEntryNode.Router",
                                                "IPv4.TorEntryNode.Directory",
                                                "IPv4.TorExitNode",
                                                "FQDN.DynDNS",
                                                "FQDN.UrlShortener",
                                                "IPv4.Bogon"
                                            ],
                                            "uniqueItems": true,
                                            "minLength": 1,
                                            "maxLength": 64
                                        },
                                        "description": "Array containing zero or more tags that can be assigned to communications by Magnet based on inferences and/or enrichments. These usually indicate the potential nature of the traffic if this might be relevant to the the information security analyst decision making.\n"
                                    }
```

This is being displayed as follows:
![screenshot](https://cloud.githubusercontent.com/assets/3287067/9591864/c9267284-5014-11e5-9ee8-180040147cf1.png)

So, the options are not being listed anywhere.

**Comment by nknapp** [comment]
It's difficult to reproduce this without having a complete swagger-spec that I can run through `bootprint`. Could you prepare a complete swagger-definition that has this bug? I would then add it to the test-suite.
(Not tonight though.)

**Comment by asieira** [comment]
Will do, thank you for the quick response.

**Comment by asieira** [comment]
Here is the full sample swagger JSON file:
```
{
    "swagger": "2.0",
    "info": {
        "title": "Test API",
        "version": "v1-beta"
    },
    "host": "api.contoso.com",
    "schemes": [
        "https"
    ],
    "basePath": "/api/v1-beta",
    "securityDefinitions": {
        "api_key": {
            "name": "X-API-KEY",
            "type": "apiKey",
            "in": "header"
        }
    },
    "security": [
        {
            "api_key": []
        }
    ],
    "consumes": [
        "application/json"
    ],
    "produces": [
        "application/json"
    ],
    "paths": {
        "/tagged_things": {
            "get": {
                "summary": "List tagged things.",
                "tags": [
                    "Alerts"
                ],
                "description": "Lists all tagged things accessible to the API key owner.\n",
                "responses": {
                    "200": {
                        "description": "An array of tagged things",
                        "schema": {
                            "type": "array",
                            "items": {
                                "type": "object",
                                "required": [
                                    "tags"
                                ],
                                "properties": {
                                    "tags": {
                                        "type": "array",
                                        "items": {
                                            "type": "string",
                                            "enum": [
                                                "option1.blah",
                                                "option2.bleh"
                                            ],
                                            "minLength": 1,
                                            "maxLength": 64
                                        },
                                        "uniqueItems": true,
                                        "description": "Array containing one or more tags.\n"
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

This is how it is displayed here:
![screenshot](https://cloud.githubusercontent.com/assets/3287067/9593432/b05b50de-5024-11e5-9804-2b1dadc1555b.png)

**Comment by nknapp** [comment]
Ah, the whole "Items" definition is missing. It's been a while, but I think I built in a shortcut (`string[]`) and omitted the Items section to not confuse the user. But this example clearly produces a wrong result.

I'll have to check  tonight, right now I have to do my daytime job...

**Comment by asieira** [comment]
It is me that has to thank you for open sourcing this amazing tools. I've tried quite a few automated documentation generators so far, and yours is the best I've seen so far. :)

**Comment by nknapp** [comment]
I've published `bootprint-json-schema@0.5.1` and `bootprint-swagger@0.6.5`. The bug was actually in  `bootprint-json-schema` which covers the schema-definition parts, but I've made a minor addition in `bootprint-swagger` so updating that as well won't hurt.

Can you verify that the bug is solved?

**Comment by asieira** [comment]
Yes, I can now confirm that the fix was effective for my swagger definition! :+1: :clap: 

Thank you very much!
